protocols/syslog/syslog.c

```diff
@@ -43,32 +45,41 @@ static Queue syslog_queue = { NULL, NULL };
 uint8_t
 syslog_send(const char *message)
 {
-  uint16_t len = strlen(message) + 1;
-  char *data = malloc(len);
+  int len = MIN(strlen(message), UIP_MAX_LENGTH);
 
+  char *data = malloc(len + 1);
   if (data == NULL)
     return 0;
+
   strncpy(data, message, len);
+  data[len] = 0;
 
-  return push(data, &syslog_queue);
+  uint8_t result = push(data, &syslog_queue);
+  if (!result)
+    free(data);
+  return result;
 }
 
 uint8_t
 syslog_sendf_P(PGM_P message, ...)
```

Comment:
```suggestion
syslog_sendf_P(PGM_P message, ...)
```
Plz change function definition to match changed declaration